### PR TITLE
upgrade to gradle 8

### PR DESCRIPTION
**Pull request by austek**
upgrade to gradle 8
upgrade depencies

### CODE REVIEW

**Comment by YOU54F** [comment]
just pushed under a branch on origin, to see if some of the failures are reltaed to being from an outside PR

https://github.com/pact-foundation/pact-plugins/tree/austek-gradle_8

**Comment by austek** [comment]
> just pushed under a branch on origin, to see if some of the failures are reltaed to being from an outside PR
> 
> https://github.com/pact-foundation/pact-plugins/tree/austek-gradle_8

that build only failed because of some Github limits 

`"message": "You have exceeded a secondary rate limit. Please wait a few minutes before you try again. If you reach out to GitHub Support for help, please include the request ID 3858:1228:3C7CB28:3CF738A:664CAE33."`

**Comment by YOU54F** [comment]
I think to get this merged you may want to at least b)

a) squash your commits
b) use conventional commit standards (so `chore(deps): update jvm driver to gradle 8`) etc as they get picked up in the changelogs. (https://github.com/pact-foundation/pact-plugins/blob/main/drivers/jvm/CHANGELOG.md)

Apart from that looks good to me, but I don't often test the consumption of the jvm-driver in pact-jvm so unsure if this may cause an issue for consumers of the jvm-driver

**Comment by YOU54F** [comment]
ty Ali!

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        jdk: [ 11, 16, 18 ]
```
Comment (rholshausen) [inline .github/workflows/gradle.yml:45]: We still need to support JDK 11

File: drivers/jvm/build.gradle [hunk 2]
```diff
@@ -1,10 +1,8 @@
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-
 plugins {
   id 'java'
-  id 'org.jetbrains.kotlin.jvm' version '1.8.22'
```
Comment (rholshausen) [inline drivers/jvm/build.gradle:5]: This will conflict with Pact-JVM libs

File: drivers/jvm/build.gradle [hunk 3]
```diff
@@ -26,8 +24,8 @@ subprojects {
 
   version = '0.4.3'
 
-  targetCompatibility = '11'
```
Comment (rholshausen) [inline drivers/jvm/build.gradle:29]: Again, this needs to be able to run on JDK 11

Comment (austek) [reply drivers/jvm/build.gradle]: `au.com.dius.pact.core:support:4.6.9` is not compatible with JDK 11

Comment (rholshausen) [reply drivers/jvm/build.gradle]: Yes, we will have to use 4.5.9

File: plugins/protobuf/build.gradle [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
     id 'application'
-    id 'com.google.protobuf' version '0.8.14'
+    id 'com.google.protobuf' version '0.9.4'
     id 'groovy'
 }
 
 version = '0.0.0'
-targetCompatibility = '11'
-sourceCompatibility = '11'
+targetCompatibility = '17'
```
Comment (YOU54F) [inline plugins/protobuf/build.gradle:12]: does this need to be 11 here with 4.5.x branch of pact-jvm?

**Comment by austek** [comment]
@rholshausen is this something you'd like to get merged, if not, I'll close the PR

**Comment by YOU54F** [comment]
Hmm has this already been cherry picked here in master

https://github.com/pact-foundation/pact-plugins/commit/5a8e1df161fe7707146ea06b31588b35e9af2d08

**Comment by rholshausen** [comment]
I thought I modified the PR and then merged. But I guess it is see it as a new commit and still labels it as unmerged.

**Comment by YOU54F** [comment]
Tres bizarre! Thanks @rholshausen and @austek 💛

**Pull request closed without merging**